test/integration/package_id/compatible_test.py

```diff
@@ -706,3 +706,70 @@ def validate_build(self):
             pkga = liba["packages"][0][pkg_index]
             assert pkga["info"]["compatibility_delta"] == {"settings": [["compiler.cppstd", "14"]]}
             assert pkga["build_args"] == "--requires=liba/0.1 --build=compatible:liba/0.1"
+
+
+def test_compatibility_new_setting_forwards_compat():
+    """ This test tries to reflect the following scenario:
+    - User adds a new setting (libc.version in this case)
+    - This setting is forward compatible
+    How is it solved with compatibility.py? Like this:
+    """
+    settings_user_file = textwrap.dedent("""
+    libc_version: [1, 2, 3, 4, 5]
+    """)
+    tc = TestClient()
+    tc.save_home({"settings_user.yml": settings_user_file})
+    tc.save({
+        "dep/conanfile.py": GenConanfile("dep", "1.0")
+            .with_settings("compiler", "libc_version"),
+        "conanfile.py": GenConanfile("app", "1.0")
+            .with_settings("compiler", "libc_version")
+            .with_require("dep/1.0")
+    })
+    tc.run("create dep -s=libc_version=4")
+    dep_package_id = tc.created_package_id("dep/1.0")
+    tc.run("create . -s=libc_version=3", assert_error=True)
+    # We can't compile, because the dep is not compatible
+    assert "Missing prebuilt package for 'dep/1.0'" in tc.out
+
+    # Let's create a compatibility extensions
+    libc_compat = textwrap.dedent("""
+    from conan.tools.scm import Version
+
+    def libc_compat(conanfile):
+        # Do we have the setting?
+        libc_version = conanfile.settings.get_safe("libc_version")
+        if libc_version is None:
+            return []
+        available_libc_versions = conanfile.settings.libc_version.possible_values()
+        ret = []
+        for possible_libc_version in available_libc_versions:
+            if Version(possible_libc_version) >= Version(libc_version):
```

Comment:
this is the other way round. Packages created with an older libc can be used with the current libc_version. So better ``<=``